integrations/nifi/nifi-rest/nifi-rest-client/nifi-rest-client-api/src/main/java/com/thinkbiganalytics/nifi/rest/client/AbstractNiFiControllerServicesRestClient.java

```diff
@@ -132,4 +303,50 @@ protected ControllerServiceDTO updateStateByIdWithRetries(@Nonnull final String
     private boolean isPendingState(@Nonnull final String currentState, @Nonnull final String finalState) {
         return ("ENABLING".equals(currentState) && "ENABLED".equals(finalState)) || ("DISABLING".equals(currentState) && "DISABLED".equals(finalState));
     }
+
+
+    private boolean ensureComponentsAreOfState(final String id, final String referenceType, final String state, final int retries, final int timeout, @Nonnull final TimeUnit timeUnit) {
+        return ensureComponentsAreOfState(id, referenceType, null, state, retries, timeout, timeUnit);
+    }
+
+
+    private boolean ensureComponentsAreOfState(final String id, final String referenceType, final Set<String> ids, final String state, final int retries, final int timeout,
+                                               @Nonnull final TimeUnit timeUnit) {
+
+        boolean allMatch = false;
+        for (int count = 0; count < retries; ++count) {
+            Optional<ControllerServiceReferencingComponentsEntity> references = getReferences(id);
+            if (references.isPresent()) {
+                allMatch =
+                    references.get().getControllerServiceReferencingComponents().stream().filter(c -> c.getComponent().getReferenceType().equalsIgnoreCase(referenceType))
+                        .filter(c -> ids != null ? ids.contains(c.getComponent().getId()) : true).allMatch(c -> state.equalsIgnoreCase(c.getComponent().getState()));
+                if (allMatch) {
+                    break;
+                } else if (!allMatch && count <= retries) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)